server/src/main/java/org/elasticsearch/cluster/routing/RoutingNode.java

```diff
@@ -67,7 +70,11 @@ private RoutingNode(RoutingNode original) {
         this.shards = new LinkedHashMap<>(original.shards);
         this.relocatingShards = new LinkedHashSet<>(original.relocatingShards);
         this.initializingShards = new LinkedHashSet<>(original.initializingShards);
+        this.startedShards = new LinkedHashSet<>(original.startedShards);
         this.shardsByIndex = Maps.copyOf(original.shardsByIndex, HashSet::new);
+        for (Map.Entry<Index, Set<ShardRouting>> entry : original.shardsByIndex.entrySet()) {
```

Comment:
is this necessary? isn't this map copied in the previous line?

Reply:
Thank you for spotting this! That's a merge artifact from one the refactoring that happened during the development of the PR. I've removed it.